conduit/pipeline/pipeline.go

```diff
@@ -329,6 +331,14 @@ func (p *pipelineImpl) Init() error {
 		go p.startMetricsServer()
 	}
 
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM, syscall.SIGINT)
+	go func() {
+		sig := <-stop
+		p.logger.Infof("Pipeline received stopping signal <%v>, stopping pipeline. p.pipelineMetadata.NextRound: %d", sig, p.pipelineMetadata.NextRound)
+		p.Stop()
+	}()
```

Comment:
You get the duplicate calls to plugins being closed because `Stop` is called twice. The other spot is in `runConduitCmdWithConfig`.

Maybe the `cli` package is the right place to install a signal handler? We pass a context into the pipeline, when the context is cancelled maybe we should implicitly call stop (or maybe cancelling the context is Stop and we get rid of the public function)

Reply:
I've converted this PR into a draft encapsulating its goals in new issue #100